.gitignore

```diff
@@ -1 +1,33 @@
-.vagrant
+# Directories
+/.vagrant
+/.idea
+/build
+
+# Files
+.DS_Store
+/*.iml
+
+# Editor swap files
+*.swp
+*.swo
+*.swn
+
+# Compiled Object files, Static and Dynamic libs (Shared Objects)
+*.o
+*.a
+*.so
+*.exe
+*.test
+*.prof
+*.pyc
+*.swp
+
+# Example binaries
+examples/df/df
+examples/df/df.exe
+examples/free/free
+examples/free/free.exe
```

Comment:
these would be already excluded by *.exe but it's nice from an "reading" perspective to have it in too.